src/scipp/core/logical.py

```diff
@@ -0,0 +1,71 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+# @author Jan-Lukas Wynen
+from __future__ import annotations
+
+from .._scipp import core as _cpp
+from ._cpp_wrapper_util import call_func as _call_cpp_func
+from ..typing import VariableLike
+
+
+def logical_not(x: VariableLike) -> VariableLike:
+    """Element-wise logical negation.
+
+    Equivalent to
+
+    .. code-block:: python
+
+        ~a
+
+    :param x: Input data.
+    :return: The logical inverse of `x`.
+    """
```

Comment:
Remember that we can use a new docstring style! But we may also want to leave this here and try to auto-convert all of them at once.

Reply:
Ah right, this was largely copy-pasted. I am going to change it, there is no reason to merge it like this.

Reply:
Done. I tried pyment but it struggles with type hints like `Union`, example sections, and see also sections. I ended up doing it with a few search and replace operations...